tooling/nargo_cli/src/cli/fv_cmd.rs

```diff
@@ -58,16 +65,259 @@ pub(crate) fn run(args: FormalVerifyCommand, workspace: Workspace) -> Result<(),
             true, // We don't want to report compile related warnings
         )?;
 
-        z3_verify(compiled_program, &workspace_file_manager, args.compile_options.deny_warnings)?
+        compiled_program
+            .ok()
+            .map(|(compiled_program, _)| {
+                z3_verify(
+                    compiled_program,
+                    &workspace_file_manager,
+                    args.compile_options.deny_warnings,
+                    &args.venir_flags,
+                )
+            })
+            .transpose()?;
```

Comment:
Maybe prefer a `.map_err(...).flat_map(...)?;` instead of `.ok().map(...).transpose()?;`

Reply:
Since we currently throw out a `Option<()>`, stemming from the original `compiled_program`